Adds Tinfoil Hats

MMMiracles opened the pull request:
:cl: MMMiracles
add: Added tinfoil hats, headgear that can help protect against government conspiracies and extra-terrestrials. Found in hacked autolathes.
/:cl:

[why]: # if you don't agree with this PR then you're part of the conspiracy

It's a hat that prevents abductors from scanning you for a disguise, makes their sleep a lot less effective, and makes you immune to the mind control mechanic from their organs. The downsides is that while making you more vulnerable to radiation and lasers, it also gives you an irreversible phobia of everything related to command and is impossible to take off without someone else's help.

MMMiracles committed:
adds a tinfoil hat to protect you from government conspiracies and aliens

Cyberboss commented on code/modules/clothing/head/misc_special.dm:
```diff
@@ -218,3 +218,10 @@
+	desc = "Thought control rays, psychotronic scanning. Don't mind that, I'm protected cause I made this hat."
+	icon_state = "foilhat"
+	item_state = "foilhat"
+	armor = list(melee = 0, bullet = 0, laser = -5,energy = 0, bomb = 0, bio = 0, rad = -5, fire = 0, acid = 0)
```
qutoes ples

MMMiracles replied:
this comment confuses me

ExcessiveUseOfCobblestone replied:
drinking at 9AM is not ok!

MMMiracles replied:
if you're asking about the quote it came from weird al's song 'foil' and if you're saying i shouldn't have a quote there then you shouldn't be part of the conspiracy

AnturK replied:
He's talking about changing `melee=` to `"melee"=`, but you can ignore it since it should be done wholesale and this is the current standard for armor lists.

Cyberboss replied:
Just do it here so I don't have to later, please

praisenarsie commented:
THE CODERS HAVE CONTROL OVER THE ADMINS!!!!!!!1

ExcessiveUseOfCobblestone commented:
other way around buddy 8)

Buggy123 commented:
Make it block ayy mind control.

>laser = -5, rad = -5

It makes you more vulnerable to lasers and radiation? That doesn't seem right.

MMMiracles commented:
@Buggy123 The joke is tinfoil is actually pretty shit at deflecting radiation in the first place and even in some cases can help direct the radiation. It also needs some downside since wearing it kinda fucks over abductors decently.

I'll see about the mind control block, that won't be difficult to throw in.

ExcessiveUseOfCobblestone commented:
lasers aren't good against aluminum so the joke is the lasers are being directed straight to your head or something.

Buggy123 commented:
>@Buggy123 The joke is tinfoil is actually pretty shit at deflecting radiation in the first place and even in some cases can help direct the radiation. It also needs some downside since wearing it kinda fucks over abductors decently.

I mean, yes I get that, but at the same time the pedantic in me says that "radiation" ingame is ionizing radiation, which would either go through tinfoil or very occasionally be absorbed by it (unless its neutron radiation, then things get _complicated_), and it would help against lasers because its shiny and reflective.

MMMiracles commented:
the hat prevents aliens from probing ur mind don't bring REALISM INTO THIS

MMMiracles committed:
qutoes ples

ExcessiveUseOfCobblestone commented:
qutoes ples

CPTANT commented:
I only support this PR if wearing a tinfoil hat gives you brain damage.

coiax commented:
Thing is, as soon as people realise there are abductors, the powergamer squad will immediately start wearing tinfoil hats.

I suggest that wearing a tinfoil hat gives you an incurable brain trauma (the one chosen should be determined from your ckey + the round ID, so you can't just take the hat on and off until you find a trauma you like), that can only be removed by removing the hat.

ExcessiveUseOfCobblestone commented:
The downside should be it influences radio waves as well as "mind control waves" meaning your headset is useless while worn.

81Denton commented:
@coiax Tbh it would be fun to have a paranoia trauma where you hallucinate being chased by aliens and security

Spaceinaba commented:
it should give you a phobia of aliens and security when you put it on

Kromgar commented:
@Spaceinaba don't forget Nanotrasen. The governments out to get me

MMMiracles committed:
In the sheets of my sterilized room
In your comfy bed, air-conditioned car
There are vapors and toxins to get you all
In the water you drink, in the air that you breath
In the soil under your shoe

MMMiracles commented:
the hats now give an irreversible phobia of security and lizards as well as being impossible to take off by yourself since why would you ever want to take it off after being shown the reality of the world you live in?

improvedname commented:
make the tinfoil hat only work when your retarded

Repukan commented:
I don't like these chemicals in the water.

It turns the friggin lizards gay

nicbn commented on code/modules/clothing/head/misc_special.dm:
```diff
@@ -218,3 +218,22 @@
+	armor = list("melee" = 0, "bullet" = 0, "laser" = -5,"energy" = 0, "bomb" = 0, "bio" = 0, "rad" = -5, "fire" = 0, "acid" = 0)
+
+/obj/item/clothing/head/foilhat/equipped(mob/living/carbon/human/user)
+	user.gain_trauma(/datum/brain_trauma/mild/phobia, FALSE, "security", "lizards")
```
What if you are security or lizard?

MMMiracles replied:
then you either don't put on the hat or you do and crumble inward as you realize everything you've ever stood for is wrong and a lie.

81Denton replied:
that's... deep

JStheguy replied:
Either that or having a loyalty implant nullifies both the conspiracy revelation and ayy blocking aspects of tinfoil hats.
The conspiracy's already got you, mate, no amount of tinfoil is going to help you now that the government's reptillian alien globalists has a whole bunch of their tech jammed into yer brain!

JJRcop commented on code/modules/clothing/head/misc_special.dm:
```diff
@@ -218,3 +218,22 @@
+	armor = list("melee" = 0, "bullet" = 0, "laser" = -5,"energy" = 0, "bomb" = 0, "bio" = 0, "rad" = -5, "fire" = 0, "acid" = 0)
+
+/obj/item/clothing/head/foilhat/equipped(mob/living/carbon/human/user)
+	user.gain_trauma(/datum/brain_trauma/mild/phobia, FALSE, "security", "lizards")
```
This actually doesn't work, the "lizards" argument is not used so this will only be security, you may want to modify the New proc of phobia to also support the `specific_type` argument being a list of phobias

MMMiracles replied:
didn't notice that, can do

coiax commented on code/modules/clothing/head/misc_special.dm:
```diff
@@ -218,3 +218,22 @@
+	user.gain_trauma(/datum/brain_trauma/mild/phobia, FALSE, "security", "lizards")
+	to_chat(user, "<span class='warning'>As you don the foiled hat, an entire world of conspiracy theories and seemingly insane ideas suddenly rush into your mind. What you once thought unbelievable suddenly seems.. undeniable. Everything is connected and nothing happens just by accident. You know too much and now they're out to get you. </span>")
+
+/obj/item/clothing/head/foilhat/attack_paw(mob/user)
```
attack_paw?

MMMiracles replied:
more beast than man

tbh i just got it from muzzle's code which stops you from taking it off yourself

JJRcop replied:
you need to do this in unequip

JJRcop commented on code/modules/clothing/head/misc_special.dm:
```diff
@@ -218,3 +218,22 @@
+	item_state = "foilhat"
+	armor = list("melee" = 0, "bullet" = 0, "laser" = -5,"energy" = 0, "bomb" = 0, "bio" = 0, "rad" = -5, "fire" = 0, "acid" = 0)
+
+/obj/item/clothing/head/foilhat/equipped(mob/living/carbon/human/user)
```
This triggers whenever you equip the hat in any slot, use the `slot` argument to see if it's equipped on the head yet
https://github.com/tgstation/tgstation/blob/8f26a27df70ffd13df88d90f8cba7e40161b6fdf/code/game/objects/items.dm#L431

WJohn commented:
Not a huge fan of the sprites. Increase the contrast between the colors rather than just make it grey in order to show that it is shiny (as metals are). Tin foil when crumpled up and folded into a hat would be full of shiny, crumpled bits where light is reflected, as well as just as many which are not, and appear nearly black by comparison.

MMMiracles committed:
conspiracy phobia, nodrop on hat once equipped, also CONTRAST

MMMiracles commented:
alright travis thanks for telling me why it failed

MimicFaux commented:
Fully support the abductor counter, but even more than that, I support giving phobias of security.
If I might make a suggestion: Forced shouting out braindamage phrases without the brain damage.

MMMiracles committed:
a single comma

CosmicScientist commented:
I *need* this.

Cyberboss commented on code/modules/clothing/head/misc_special.dm:
```diff
@@ -218,3 +218,16 @@
+	if(slot == slot_head)
+		user.gain_trauma(/datum/brain_trauma/mild/phobia, FALSE, "conspiracies")
+		to_chat(user, "<span class='warning'>As you don the foiled hat, an entire world of conspiracy theories and seemingly insane ideas suddenly rush into your mind. What you once thought unbelievable suddenly seems.. undeniable. Everything is connected and nothing happens just by accident. You know too much and now they're out to get you. </span>")
+		src.flags_1 += NODROP_1
```
remove src

Cyberboss commented on code/modules/antagonists/abductor/machinery/console.dm:
```diff
@@ -166,6 +166,10 @@
 
 /obj/machinery/abductor/console/proc/AddSnapshot(mob/living/carbon/human/target)
+	var/obj/O = target.get_item_by_slot(slot_head)
+	if(istype(O, /obj/item/clothing/head/foilhat))
```
The O var is unecessary

Cyberboss commented on code/modules/antagonists/abductor/equipment/abduction_gear.dm:
```diff
@@ -316,6 +316,7 @@
 	if(iscarbon(target))
 		var/mob/living/carbon/C = target
 		var/obj/item/organ/heart/gland/G = C.getorganslot("heart")
+		var/obj/O = C.get_item_by_slot(slot_head)
```
The O var is unecessary

MMMiracles replied:
Is there a way to do this while still being able to reference the hat item for the messages?

ShizCalev replied:
if(istype(C.get_item_by_slot(slot_head), /obj/item/clothing/head/foilhat))

MMMiracles replied:
ohhh, i forgot the istype when I tried that initially. thanks

Kerbin-Fiber commented:
Make it impossible for you to take off someone else's tinfoil hat if you're wearing one too since you don't want to expose them to the world you just discovered either.

Poojawa commented:
It wouldn't really help because "lasers" in this code have recoil and slower than light speeds 

thus. particle blasters of somesort

Cyberboss committed:
Merge branch 'master' into conspiracy_but_true_question_mark

CosmicScientist commented:
WHEEEEN

MMMiracles committed:
cyberboss changes

MMMiracles committed:
Merge branch 'conspiracy_but_true_question_mark' of https://github.com/MMMiracles/-tg-station into conspiracy_but_true_question_mark

optimumtact commented on code/modules/clothing/head/misc_special.dm:
```diff
@@ -218,3 +218,16 @@
+	if(slot == slot_head)
+		user.gain_trauma(/datum/brain_trauma/mild/phobia, FALSE, "conspiracies")
+		to_chat(user, "<span class='warning'>As you don the foiled hat, an entire world of conspiracy theories and seemingly insane ideas suddenly rush into your mind. What you once thought unbelievable suddenly seems.. undeniable. Everything is connected and nothing happens just by accident. You know too much and now they're out to get you. </span>")
+		flags_1 += NODROP_1
```
this is not how you do bitflags

optimumtact replied:
Also no drop prevents other people stripping the item afaik

optimumtact replied:
https://tgstation13.org/wiki/Binary_flags

optimumtact replied:
I think there is a cursed flag that will let others strip an item off you, can't quite remember though

JJRcop replied:
I think the best way to do this would be to create a mob_can_unequip proc for items, have that proc handle the nodrop flag, and then either create a cursed flag or directly override the foil hat's mob_can_unequip proc

CosmicScientist commented:
WHEN

Cyberboss commented:
When it's fixed

On Feb 25, 2018 7:40 AM, "CosmicScientist" <notifications@github.com> wrote:

> WHEN
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/35665#issuecomment-368306093>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHywem6hwEsexByl8GNWdHI2M2rHsQF5ks5tYVS_gaJpZM4SFWeV>
> .
>

MMMiracles commented:
sorry i got caught up elsewhere i'll get the bit flag fixed now before the commit skeleton comes to rattle their bones

MMMiracles committed:
|=

optimumtact commented on code/modules/clothing/head/misc_special.dm:
```diff
@@ -218,3 +218,16 @@
+	if(slot == slot_head)
+		user.gain_trauma(/datum/brain_trauma/mild/phobia, FALSE, "conspiracies")
+		to_chat(user, "<span class='warning'>As you don the foiled hat, an entire world of conspiracy theories and seemingly insane ideas suddenly rush into your mind. What you once thought unbelievable suddenly seems.. undeniable. Everything is connected and nothing happens just by accident. You know too much and now they're out to get you. </span>")
+		flags_1 |= NODROP_1
```
they are never going to be able to take it off, which might be what you want, I don't know

MMMiracles replied:
the intention is that they can't remove it once they wear it, yeah.

optimumtact commented on strings/phobia.json:
```diff
@@ -25,6 +25,42 @@
 	],
+	
+	"conspiracies": [
+		"cerntral command",
```
cerntral?

MMMiracles committed:
central typo fix

CosmicScientist commented:
eeeeeeeeeeeeeee

Memelizard commented:
Pretty much keeps you permapanicked/feared if you put one on as  a lizard. Might want to remove the fear of lizards from them.

Repukan commented:
Don't put them on then you idiot

Memelizard commented:
Doesn't work if it's forced on you.

Also it makes mimes talk.
